src/django/api/views.py

```diff
@@ -3222,7 +3222,8 @@ def validate_request(self, request):
 
     def list(self, request):
         self.validate_request(request)
-        response_data = ApiBlockSerializer(self.queryset, many=True).data
+        response_data = ApiBlockSerializer(
+            ApiBlock.objects.all(), many=True).data
```

Comment:
I think this is ok since we documented it with a comment message but it surprised me that we needed to do this. I explored plain Django queryset interactions in `shell_plus` and verified my expectations that each evaluation of a queryset would fetch new results.

```
>>> q = ApiBlock.objects.all()
>>> q
<QuerySet [<ApiBlock: ApiBlock 1 - Contributor 2 until 2020-11-30 00:00:00+00:00>]>
>>> b = ApiBlock.objects.get(id=1)
>>> b.until = '2020-11-29'
>>> b.save()
>>> q
<QuerySet [<ApiBlock: ApiBlock 1 - Contributor 2 until 2020-11-29 00:00:00+00:00>]>
```

I don't know how DRF is altering this behavior. Is there some documentation that pointed you to this solution?

Reply:
Only a few StackOverflow posts. [This one](https://stackoverflow.com/a/42764200) suggested that the class's queryset is set once on initialization, and then isn't refreshed. "...anything happening outside a function (ie at a module's top-level, in a class statement's body etc) is executed only once (per process) at module's first import". Switching to fetching the data directly in the `list` request then fixed the issue, so it seems to be accurate.